internal/provider/data_file.go

```diff
@@ -0,0 +1,146 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/neuspaces/terraform-provider-system/internal/client"
+	"github.com/neuspaces/terraform-provider-system/internal/validate"
+	"path"
+)
+
+const dataFileName = "system_file"
+
+func dataFile() *schema.Resource {
+	return &schema.Resource{
+		Description: fmt.Sprintf("`%s` retrieves information about a file on the remote system.", dataFileName),
+
+		ReadContext: dataFileRead,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		SchemaVersion: 1,
+
+		Schema: map[string]*schema.Schema{
+			resourceFileAttrId: {
+				Description: "ID of the file",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			resourceFileAttrPath: {
+				Description:      "Path to the file. Must be an absolute path.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validate.AbsolutePath(),
```

Comment:
1 error occurred:
        	            		* data source system_file: mode: ValidateDiagFunc is for validating user input, there's nothing to validate on computed-only field

Reply:
ok, removed validation